homeassistant/components/comelit/switch.py

```diff
@@ -4,41 +4,123 @@
 
 from typing import Any
 
-from aiocomelit import ComelitSerialBridgeObject
-from aiocomelit.const import IRRIGATION, OTHER, STATE_OFF, STATE_ON
+from aiocomelit import ComelitSerialBridgeObject, ComelitVedoZoneObject
+from aiocomelit.const import (
+    ALARM_ZONES,
+    BRIDGE,
+    IRRIGATION,
+    OTHER,
+    STATE_OFF,
+    STATE_ON,
+    AlarmZoneState,
+)
 
 from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
 from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_TYPE
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
 from .const import DOMAIN
-from .coordinator import ComelitSerialBridge
+from .coordinator import ComelitSerialBridge, ComelitVedoSystem
 
 
 async def async_setup_entry(
     hass: HomeAssistant,
     config_entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Set up Comelit switches."""
+    """Set up Comelit sensors."""
+
+    if config_entry.data.get(CONF_TYPE, BRIDGE) == BRIDGE:
+        await async_setup_bridge_entry(hass, config_entry, async_add_entities)
+    else:
+        await async_setup_vedo_entry(hass, config_entry, async_add_entities)
+
+
+async def async_setup_bridge_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Comelit Bridge switches."""
 
     coordinator: ComelitSerialBridge = hass.data[DOMAIN][config_entry.entry_id]
 
-    entities: list[ComelitSwitchEntity] = []
+    entities: list[ComelitBridgeSwitchEntity] = []
     entities.extend(
-        ComelitSwitchEntity(coordinator, device, config_entry.entry_id)
+        ComelitBridgeSwitchEntity(coordinator, device, config_entry.entry_id)
         for device in coordinator.data[IRRIGATION].values()
     )
     entities.extend(
-        ComelitSwitchEntity(coordinator, device, config_entry.entry_id)
+        ComelitBridgeSwitchEntity(coordinator, device, config_entry.entry_id)
         for device in coordinator.data[OTHER].values()
     )
     async_add_entities(entities)
 
 
-class ComelitSwitchEntity(CoordinatorEntity[ComelitSerialBridge], SwitchEntity):
```

Comment:
I think it would be better to switch to `EntityDescription `instead of duplicating code

Reply:
Most of the code is identical, we should use a different approach.